### ISSUE TITLE
Does not work on Standoff 2 0.31.1

### ISSUE THREAD

**Issue by NoPref** [Michel-M-code/Metadata-Decryptor#8]
It is throws an error at 85%

```
Select mode. Automatic or Manual: a
Automatic mode selected.
Input encrypted metadata file path: C:\Users\####\Downloads\new\global-metadata.dat
Input reference metadata file path: C:\Users\####\Downloads\global-metadata.dat
Input decrypted metadata save path: decrypted.dat
Using next files:
    Encrypted - C:\Users\####\Downloads\new\global-metadata.dat
    Reference - C:\Users\####\Downloads\global-metadata.dat
    Output - decrypted.dat
Correct? 1
Searching... : 100%|███████████████████████████████████████| 1922/1922 [00:01<00:00, 1588.24pairs/s]
Reordering...:  85%|██████████████████████████████▋     | 1025/1204 [00:00<00:00, 1144307.05pairs/s]
DECRYPTION FAILED!
```

However the version 0.31.0 decryptes without an error

**Comment by NoPref** [comment]
Thanks for the reply! Honestly, I don't understand what the problem is. I took the metadata from 0.31.1 and for reference 0.27.3 (I tried to take apk from different sites: apkpure, apkmirror, pdalife). It didn't work. I decided to try to run the script on vds. The result is the same. If I change the wrong_count in the script from 30 to 36, the metadata is decrypted without problems, but I can't use it in il2cppdumper because of an error:

Initializing metadata...
System.ArgumentException: An item with the same key has already been added. Key: 35912
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey , TValue , InsertionBehavior )
   at Il2CppDumper.Metadata..ctor(Stream stream) in C:\projects\il2cppdumper\Il2CppDumper\Il2Cpp\Metadata.cs:line 145
   at Il2CppDumper.Program.Init(String il2cppPath, String metadataPath, Metadata& metadata, Il2Cpp& il2Cpp) in C:\projects\il2cppdumper\Il2CppDumper\Program.cs:line 124

When I have time I will try to add more logging to the script and will post the results

[screen from the vds]

![{6AD786B6-134E-46FC-B2C8-74C84DEFDD04}](https://github.com/user-attachments/assets/343d537a-21b5-4a40-ac54-dd8dd37da404)
